test/cmp_hdr_test.c

```diff
@@ -137,8 +137,10 @@ static int execute_HDR_set1_recipient_test(CMP_HDR_TEST_FIXTURE *fixture)
         return 0;
 
     X509_NAME_ADD(x509name, "CN", "A common recipient name");
-    if (!TEST_int_eq(ossl_cmp_hdr_set1_recipient(fixture->hdr, x509name), 1))
+    if (!TEST_int_eq(ossl_cmp_hdr_set1_recipient(fixture->hdr, x509name), 1)) {
+        X509_NAME_free(x509name);
```

Comment:
Doesnt this have to be ddone for evvery return zero in this function?  i.e. perhaps you should make the return zeros a goto out, where the X509_NAME_free is already called, and then just return a stack variable properly set to 1 or 0?

Reply:
Like this?

```
static int execute_HDR_set1_recipient_test(CMP_HDR_TEST_FIXTURE *fixture)
{
    X509_NAME *x509name = X509_NAME_new();
    int res = 0;

    if (!TEST_ptr(x509name))
        return 0;

    X509_NAME_ADD(x509name, "CN", "A common recipient name");
    if (!TEST_int_eq(ossl_cmp_hdr_set1_recipient(fixture->hdr, x509name), 1))
        goto err;

    if (!TEST_int_eq(fixture->hdr->recipient->type, GEN_DIRNAME))
        goto err;

    if (!TEST_int_eq(X509_NAME_cmp(fixture->hdr->recipient->d.directoryName,
                                   x509name), 0))
        goto err;

    res = 1;

 err:
    X509_NAME_free(x509name);
    return res;
}
```